main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1871,6 +1871,16 @@ object Typer {
           resultEff = Type.mkUnion(eff1, eff2, loc)
         } yield (constrs1 ++ constrs2, resultTyp, resultPur, resultEff)
 
+      case KindedAst.Expression.Error(m, tvar, pvar, evar) =>
+        val constrs = Nil
+        val expectedType = Type.freshVar(Kind.Star, m.loc)
+        val expectedPur = Type.freshVar(Kind.Bool, m.loc)
+        val expectedEff = Type.freshVar(Kind.Effect, m.loc)
+        for {
+          resultTyp <- unifyTypeM(tvar, expectedType, m.loc)
+          resultPur <- unifyTypeM(pvar, expectedPur, m.loc)
+          resultEff <- unifyTypeM(evar, expectedEff, m.loc)
+        } yield (constrs, resultTyp, resultPur, resultEff)
```

Comment:
Actually, you dont need `expectedType`, and friends. You can simply return `(Nil, tvar, pvar, evar)`.